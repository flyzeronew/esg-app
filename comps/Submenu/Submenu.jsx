import styles from "./Submenu.module.css";
import classNames from "classnames/bind";

const cx = classNames.bind(styles)

function Submenu(props) {
    return (
        <>
            <div className={cx("submenuArea")}>
                <div className={cx("submenu")}>
                    <div className={cx("submenuMask")}></div>
                    <a href={`/${props.page}`} className={!props.genreId ? cx("act") : ''}  >全部</a>
                    {props.submenu?.length > 0 ? props.submenu.map((item, index) => (
                        <a
                            key={index}
                            className={item.en_name == props.genreEnName ? cx("act") : ''}
                            href={`/${props.page}/${item.en_name}`}
                        >
                            {item.name}
                        </a>
                    )) : ''
                    }
                </div>
            </div>
            <div className={cx("moSubmenuArea")}>
                <div>
                <a href={`/${props.page}`} className={!props.genreId ? cx("act") : ''} >全部</a>
                </div>
                {props.submenu?.length > 0 ? props.submenu.map((item, index) => (
                    <div key={"moSubmenuArea" + index}>
                        <a
                            key={index}
                            className={item.en_name == props.genreEnName ? cx('act') : ''}
                            href={`/${props.page}/${item.en_name}`}
                        >
                            {item.name}
                        </a>
                    </div>
                )) : ''
                }
            </div>
        </>
    )
}
export default Submenu;