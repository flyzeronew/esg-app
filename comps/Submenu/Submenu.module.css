.submenu{
    margin:0 32px 50px;
    padding:10px 32px;
    white-space:nowrap;
    height: max-content;
    font-family: var(--font-family-noto);
}
.submenu a{
    margin-right:16px;
    padding: 8px 12px;
    border-radius: 50px;
    background-color: #F2F2F2;
    transition: 0.3s;
    color: #333333;
    font-size: var(--font-size-lg);
}
.submenu a.act{
    background-color: #00D780;
}
.moSubmenuArea {
    display: none;
}
@media (max-width:1023px) { 
    .submenuArea{
        display: none;
    }
    .moSubmenuArea {
        display: flex;
        justify-content: center;
        flex-flow:wrap;
        gap: 24px 16px;
        padding:10px;
        margin: 14px;
        margin:14px 14px 32px;
        white-space:nowrap;
        height: max-content;
        font-family: var(--font-family-noto);
    }
    .moSubmenuArea a{
        padding: 8px 12px;
        border-radius: 50px;
        background-color: #F2F2F2;
        transition: 0.3s;
        color: #333333;
        font-size: var(--font-size-lg);
    }
    .moSubmenuArea a.act{
        background-color: #00D780;
    }
}