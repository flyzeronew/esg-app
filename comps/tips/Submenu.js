import React, { Component } from "react"
function Submenu(props) {
    const appUrl = process.env.APP_URL;
    return (        
        <div className='submenuArea'>
            <div className='submenu'> 
                <div className='submenuMask'></div>
                    <a href={`${appUrl}/tips`} className={!props.genreId ? 'act':''} >全部</a>
                        {props.submenu.length > 0 ? props.submenu.map((item, index) => (
                                <a 
                                    key={index} 
                                    className={ index+1 == props.genreId ? 'act':''} 
                                    href={`${appUrl}/tips/${item.en_name}`}
                                >
                                    {item.name}
                                </a>
                            )):''
                    }
            </div>
        </div>
    )
}
export default Submenu;