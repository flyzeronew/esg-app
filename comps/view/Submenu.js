import React, { Component } from "react"

function Submenu(props) {
    const appUrl = process.env.APP_URL;
    return (        
        <div className='submenuArea'>

            <div className='submenu'> 
                <div className='submenuMask'></div>
                    <a href={`${appUrl}/view`} className={!props.genreId ? 'act':''} >全部</a>
                        {props.submenu.length > 0 ? props.submenu.map((item, index) => (
                                <a 
                                    key={index} 
                                    className={ index+1 == props.genreId ? 'act':''} 
                                    href={`${appUrl}/view/genres/${item.id}`}
                                >
                                    {item.name}
                                </a>
                            )):''
                    }
            </div>
        </div>
    )
}
export default Submenu;