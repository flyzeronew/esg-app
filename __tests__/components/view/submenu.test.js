import { render, screen, logRoles } from '@testing-library/react';
import Submenu from '@/comps/view/Submenu';


process.env.IMG_DEFAULT = '/default-esg-image.jpg';

describe("Submenu component view pages", ()=>{
    const mockSumenu = [
        {
          "id": 1,
          "name": "主題企劃",
          "en_name": "trend",
          "description": "主題式內容帶你探索全球永續熱議話題，透過專題文章，了解友善環境、氣候變遷、綠色生活等議題的重要性，讓永續走入我們的日常，探索未來生活的樣貌。"
        },
        {
          "id": 2,
          "name": "人物專訪",
          "en_name": "interviews",
          "description": "聚焦各行各業的永續推動者，透過品牌創辦人與實踐者的真實故事，分享他們的理念、挑戰與成就。讓你發現永續其實蘊藏在生活中，啟發每個人踏出行動的第一步！"
        }
      ];
    
    it("submenu unchanged", ()=>{
        const {container} = render(<Submenu submenu={mockSumenu}></Submenu>);
        expect(container).toMatchSnapshot();
    })
    it("Renders submenu without any data",()=>{
        render(<Submenu submenu={[]}></Submenu>);
        expect(screen.getAllByText("全部")[0]).toHaveClass("act");
        expect(screen.queryAllByLabelText("人物專訪")).toHaveLength(0)
    })
    it("renders submenu with data", ()=>{
        render(<Submenu submenu={mockSumenu}></Submenu>);
        expect(screen.getAllByText("人物專訪")[0]).toBeInTheDocument();
        expect(screen.getAllByText("人物專訪")[0]).toHaveAttribute('href', '/view/interviews');
    })
    it("Renders submenu with Genre",()=>{
        render(<Submenu submenu={mockSumenu} genreEnName={"interviews"}></Submenu>);
        expect(screen.getAllByText("人物專訪")[0]).toHaveClass("act");
    })
  
})