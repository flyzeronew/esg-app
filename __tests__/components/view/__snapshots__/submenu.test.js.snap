// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submenu component view pages submenu unchanged 1`] = `
<div>
  <div
    class="submenuArea"
  >
    <div
      class="submenu"
    >
      <div
        class="submenuMask"
      />
      <a
        class="act"
        href="/view"
      >
        全部
      </a>
      <a
        class=""
        href="/view/trend"
      >
        主題企劃
      </a>
      <a
        class=""
        href="/view/interviews"
      >
        人物專訪
      </a>
    </div>
  </div>
  <div
    class="moSubmenuArea"
  >
    <div>
      <a
        class="act"
        href="/view"
      >
        全部
      </a>
    </div>
    <div>
      <a
        class=""
        href="/view/trend"
      >
        主題企劃
      </a>
    </div>
    <div>
      <a
        class=""
        href="/view/interviews"
      >
        人物專訪
      </a>
    </div>
  </div>
</div>
`;
